datafusion/core/src/physical_plan/aggregates/mod.rs

```diff
@@ -90,18 +121,24 @@ impl AggregateExec {
     /// Create a new hash aggregate execution plan
     pub fn try_new(
         mode: AggregateMode,
-        group_expr: Vec<(Arc<dyn PhysicalExpr>, String)>,
+        grouping_set: PhysicalGroupingSet,
         aggr_expr: Vec<Arc<dyn AggregateExpr>>,
         input: Arc<dyn ExecutionPlan>,
         input_schema: SchemaRef,
     ) -> Result<Self> {
-        let schema = create_schema(&input.schema(), &group_expr, &aggr_expr, mode)?;
+        let schema = create_schema(
+            &input.schema(),
+            &grouping_set.expr,
+            &aggr_expr,
+            grouping_set.groups.iter().flatten().any(|is_null| *is_null),
```

Comment:
I wonder if extracting this code to a function such as  `GroupingSets::contains_null()` might make the code easier to read. The same comment applies to other places where `GroupingSets::groups` is referenced as well. 

Given the size of this PR already, definitely could be done as a follow on